src/Stratis.Bitcoin/P2P/Peer/NetworkPeerConnection.cs

```diff
@@ -149,15 +149,18 @@ private async Task ReceiveMessagesAsync()
                     this.MessageProducer.PushMessage(incomingMessage);
                 }
             }
-            catch (OperationCanceledException)
-            {
-                this.logger.LogTrace("Receiving cancelled.");
-                this.peer.Disconnect("Receiving cancelled.");
-            }
             catch (Exception ex)
             {
-                this.logger.LogTrace("Exception occurred: '{0}'", ex.ToString());
-                this.peer.Disconnect("Unexpected failure while waiting for a message", ex);
+                if ((ex is IOException) || (ex is OperationCanceledException) || (ex is ObjectDisposedException))
```

Comment:
Since we are expecting those exceptions we need to update xmldoc for the method that can throw. Right now `ReadAndParseMessageAsync` throws only OperationCanceledException and FormatException, `PushMessage` throws nothing.